packages/ts-morph/src/tests/compiler/ast/jsx/jsxElementTests.ts

```diff
@@ -159,4 +159,34 @@ const v = <div>
       });
     });
   });
+
+  describe(nameof<JsxElement>("remove"), () => {
+    function doRemove(text: string) {
+      const { descendant, sourceFile } = getInfo(text);
+      descendant.remove();
+    }
+
+    function doTestWithJsxElementChild(text: string, expected: string) {
+      const { descendant, sourceFile } = getInfo(text);
+      (descendant.getFirstDescendantByKind(SyntaxKind.JsxElement) as JsxElement).remove();
+      expect(sourceFile.getFullText()).to.equal(expected);
+    }
+
+    it("should not remove the root JsxElement", () => {
+      let error = null;
+
+      try {
+        doRemove(`var t = (<jsx></jsx>);`);
+      }
+      catch (err) {
+        error = err;
+      }
+
+      expect(error).to.be.instanceOf(errors.InvalidOperationError);
+    });
+
+    it("should remove the JsxElement child", () => {
+      doTestWithJsxElementChild(`var t = (<jsx><jsx2></jsx2></jsx>);`, `var t = (<jsx></jsx>);`);
```

Comment:
It's not so simple. Once spaces are added then it gets much more complicated because spaces have meaning in JSX. It's why I haven't had the time to add this functionality yet. For example:

```ts
it("should remove the JsxElement child when has spaces", () => {
      doTestWithJsxElementChild(
        `var t = (<jsx>
        <jsx2></jsx2>
      </jsx>);`,
        `var t = (<jsx></jsx>);`,
      );
    });
```